astropy/table/index.py

```diff
@@ -594,11 +586,14 @@ def get_slice(self, col_slice, item):
         '''
         from .table import Table
         if len(self.columns) == 1:
-            return Index([col_slice], engine=self.data.__class__)
+            index = Index([col_slice], engine=self.data.__class__)
+            return SlicedIndex(index, slice(0, 0, None), original=True)
```

Comment:
Can't really judge this, but it certainly seems more logical. Maybe even more logical would be `self.__class__(index, ...)`

Reply:
Done.